internal/gatewayapi/validate.go

```diff
@@ -240,111 +241,126 @@ func (t *Translator) validateTLSConfiguration(listener *ListenerContext, resourc
 			break
 		}
 
-		if len(listener.TLS.CertificateRefs) != 1 {
+		if len(listener.TLS.CertificateRefs) == 0 {
 			listener.SetCondition(
 				v1beta1.ListenerConditionProgrammed,
 				metav1.ConditionFalse,
 				v1beta1.ListenerReasonInvalid,
-				"Listener must have exactly 1 TLS certificate ref",
+				"Listener must have at least 1 TLS certificate ref",
 			)
 			break
 		}
 
-		certificateRef := listener.TLS.CertificateRefs[0]
+		pkaSecretSet := make(map[string]*v1.Secret)
+		for _, certificateRef := range listener.TLS.CertificateRefs {
+			if certificateRef.Group != nil && string(*certificateRef.Group) != "" {
+				listener.SetCondition(
+					v1beta1.ListenerConditionResolvedRefs,
+					metav1.ConditionFalse,
+					v1beta1.ListenerReasonInvalidCertificateRef,
+					"Listener's TLS certificate ref group must be unspecified/empty.",
+				)
+				break
+			}
 
-		if certificateRef.Group != nil && string(*certificateRef.Group) != "" {
-			listener.SetCondition(
-				v1beta1.ListenerConditionResolvedRefs,
-				metav1.ConditionFalse,
-				v1beta1.ListenerReasonInvalidCertificateRef,
-				"Listener's TLS certificate ref group must be unspecified/empty.",
-			)
-			break
-		}
+			if certificateRef.Kind != nil && string(*certificateRef.Kind) != KindSecret {
+				listener.SetCondition(
+					v1beta1.ListenerConditionResolvedRefs,
+					metav1.ConditionFalse,
+					v1beta1.ListenerReasonInvalidCertificateRef,
+					fmt.Sprintf("Listener's TLS certificate ref kind must be %s.", KindSecret),
```

Comment:
maybe also add an index into this string so the user knows which certRef is invalid

Reply:
This is before we're checking the namespace name. Would an index be alright here?

Reply:
yes if there are two invalid `Kind`s, it will help decipher which one the status is referring to